components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,512 @@
+		for _, lowerMatchId in ipairs(firstBracketData.lowerMatchIds) do
+			sfMatchIds[lowerMatchId] = true
+		end
+	end
+	return sfMatchIds
+end
+
+-- Finds the first round in where upper bracket opponents drop to the lower
+-- bracket. Returns nil if it cannot be determined unambiguously, or if the
+-- bracket is not double elimination.
+function Import._findDeRoundIndex(bracket)
+	if #bracket.sections ~= 2 then
+		return
+	end
+	local countsByRound = MatchGroupCoordinates.computeRawCounts(bracket)
+
+	for roundIndex = 1, #bracket.rounds do
+		local lbCount = countsByRound[roundIndex][2]
+		if lbCount == 0 then
+			return roundIndex
+		elseif lbCount > 0 then
+			return
+		end
+	end
+end
+
+function Import._mergePlacements(lpdbEntries, placements)
+	for placementIndex, lpdbPlacement in ipairs(lpdbEntries) do
+		placements[placementIndex] = Import._mergePlacement(
+			lpdbPlacement,
+			placements[placementIndex] or Import._emptyPlacement(placements[placementIndex - 1], #lpdbPlacement)
+		)
+	end
+
+	return placements
+end
+
+function Import._emptyPlacement(priorPlacement, placementSize)
+	priorPlacement = priorPlacement or {}
+	local placeStart = (priorPlacement.placeEnd or 0) + 1
+	local placeEnd = (priorPlacement.placeEnd or 0) + placementSize
+
+	return Placement(
+		{placeStart = placeStart, placeEnd = placeEnd},
+		priorPlacement.parent,
+		priorPlacement.placeEnd or 0
+	)
+end
+
+function Import._mergePlacement(lpdbEntries, placement)
+	for opponentIndex, opponent in ipairs(lpdbEntries) do
+		placement.opponents[opponentIndex] = Import._mergeEntry(
+			opponent,
+			Table.mergeInto(placement:_parseOpponents{{}}[1], placement.opponents[opponentIndex]),
+			placement
+		)
+	end
+
+	assert(
+		#placement.opponents <= 1 + placement.placeEnd - placement.placeStart,
+		'Import: Too many opponents returned from query for placement range '
+			.. placement:_displayPlace():gsub('&#045;', '-')
+	)
+
+	return placement
+end
+
+function Import._mergeEntry(lpdbEntry, entry, placement)
+	if
+		not Opponent.isTbd(entry.opponentData) -- valid manual input
+		or Table.isEmpty(lpdbEntry.opponent) -- irrelevant lpdbEntry
+		or Opponent.isTbd(lpdbEntry.opponent)
+	then
+		return entry
+	end
+
+	return Table.deepMergeInto(Import._entryToOpponent(lpdbEntry, placement), Import._removeTbd(entry))
+end
+
+function Import._removeTbd(entry)
+	-- if we get here we know entry is a TBD Opponent
+	-- but it still might hold data (e.g. flag or race of players) for non party opponents
+	local opponent = entry.opponentData
+	if not Opponent.typeIsParty(opponent.type) then
+		-- TBD opponents of non party types do not hold additional data
+		entry.opponentData = {}
+	else
+		opponent.type = nil
+		for _, player in pairs(opponent.players or {}) do
+			if Opponent.playerIsTbd(player) then
+				player.displayName = nil
+				player.pageIsResolved = nil
+				player.pageName = nil
+			end
+		end
+	end
+
+	return entry
+end
```

Comment:
```suggestion
function Import._removeTbd(opponent)
	if not Opponent.isTbd(opponent) then
		return opponent
	end

	-- Entry is a TBD Opponent, but could contain data (e.g. flag or faction) for non party opponents
	if not opponent.type or not Opponent.typeIsParty(opponent.type) then
		return {}
	else
		opponent.type = nil
		for _, player in pairs(opponent.players or {}) do
			if Opponent.playerIsTbd(player) then
				player.displayName = nil
				player.pageIsResolved = nil
				player.pageName = nil
			end
		end
	end

	return opponent
end
```

Would this work as a merge between `_removeTbd` and `_removeIfTbd`? https://github.com/Liquipedia/Lua-Modules/pull/1803/files#diff-6d536082818f6da85f885d2a33d40bd81b4f7baaf6d9adb13f81709819248ac2R387 would need to be updated also

Reply:
will need to check on pc in the evening but i don't see how this should work for the removeIfTbd in the case of party opponents, which returns an empty table if it is a tbd opponent, while this returns an opponent with removed names and displaynames

Reply:
nope it just breaks the merge

Reply:
maybe rename `_removeIfTbd` to `_emptyIfTbd` and `_removeTbd` to `_removeTbdIdentifiers`?

Reply:
adjusted stuff further and merged the functions